containerd-shim/process_linux.go

```diff
@@ -0,0 +1,96 @@
+// +build !solaris
+
+package main
+
+import (
+	"io"
+	"os"
+	"syscall"
+)
+
+// setPDeathSig sets the parent death signal to SIGKILL so that if the
+// shim dies the container process also dies.
+func setPDeathSig() *syscall.SysProcAttr {
+	return &syscall.SysProcAttr{
+		Pdeathsig: syscall.SIGKILL,
+	}
+}
+
+// openIO opens the pre-created fifo's for use with the container
+// in RDWR so that they remain open if the other side stops listening
+func (p *process) openIO() error {
```

Comment:
This should be updated to use the version added by @tonistiigi  (see [here](https://github.com/docker/containerd/pull/330))